feat(client): v1 python library

besaleli opened the pull request:
# Description
MVP for V1 Python Library!

Additionally:
- Updated Makefile
- Added Codspeed for Python benchmarking
- Updated formatting to include more relevant python dirs

**Linked Issue:** Closes #299
**Type of Change:** ✨ feat

# Testing
Added unit  tests for critical validation functionality. We need to figure out how to mock gRPC stubs for queueing. Will add integration tests once we solidify v1 a bit more.

# Checklist
- [x] My code follows conventional commit guidelines
- [x] I have added tests that prove my fix/feature works
- [x] New and existing tests pass locally
- [x] I have updated relevant documentation
- [x] I have added metrics/monitoring for new features (if applicable)
```

pachewise commented on ptolemy-py/python/ptolemy_client/v1/io.py:
```diff
@@ -0,0 +1,60 @@
+    error_type: Optional[str] = Field(default=None)
+    error_content: Optional[str] = Field(default=None)
+
+    def start(self):
+        """Start runtime log."""
+
+        if self.start_time is not None:
+            raise ValueError("Runtime already started.")
+
+        self.start_time = time.time()
```
Hmmm - i think Runtime can just be a data object here; Since you are calling this in the parent Trace object, why not just have the trace object create this Runtime dataclass with the right start_time/end_time (and it can be unchangeable after thaT)

pachewise reviewed (commented):
first pass - haven't really looked at the Rust code; but on the Python code, my main feedback is you should prob have "factory" objects and "data" objects, with a clear separator for both. Will simplify your Pydantic usage, and may make the Rust-Python bindings simpler (🤷🏾 maybe will simplify those to_record calls?).

pachewise commented on ptolemy-py/python/ptolemy_client/v1/client.py:
```diff
@@ -94,6 +55,20 @@ def trace(
+
+    return Ptolemy(
+        base_url=base_url,
+        client=RecordExporter(base_url),
```
shouldn't you just use the client instance you created before?

pachewise replied:
<img width="531" height="200" alt="Image" src="https://github.com/user-attachments/assets/2e7fa0ad-edf6-4d8b-88e7-2ec4ab399105" />

pachewise commented on ptolemy-py/python/ptolemy_client/v1/client.py:
```diff
@@ -0,0 +1,207 @@
+    tier: Tier
+    name: str
+
+    parameters: Optional[Parameters] = Field(default=None)
+
+    version: Optional[str] = Field(default=None)
+    environment: Optional[str] = Field(default=None)
+
+    start_time: Optional[float] = None
+
+    runtime_: Optional[Runtime] = Field(default=None)
+
+    inputs_: Optional[List[IO[Any]]] = Field(default=None)
+    outputs_: Optional[List[IO[Any]]] = Field(default=None)
+    feedback_: Optional[List[IO[Any]]] = Field(default=None)
+    metadata_: Optional[List[IO[str]]] = Field(default=None)
```
when would these be null?

besaleli commented:
@pachewise to addressing your comments (in order):
1) what client? it's not initialized globally
2) inputs/outputs/etc will be null before they're set.
